### PR TITLE
Remove deprecated syntax

### PR DESCRIPTION
**Pull request by alpaca-tc**
`alias_method_chain` is deprecated, replaced by `Module.prepend`.
Only thing I am unsure of is legacy support.

### CODE REVIEW

File: lib/test_after_commit.rb [hunk 1]
```diff
@@ -1,5 +1,13 @@
 require 'test_after_commit/version'
 
+if ActiveRecord::VERSION::MAJOR >= 4
+  require 'test_after_commit/with_transaction_state'
+  ActiveRecord::Base.__send__(:prepend, TestAfterCommit::WithTransactionState)
```
Comment (grosser) [inline lib/test_after_commit.rb]: why the __send__ ... it's ruby 2.2 so a `.prepend` would do ?

File: lib/test_after_commit/with_transaction_state.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# disable parts of the sync code that starts looping
+module TestAfterCommit
+  module WithTransactionState
+    def sync_with_transaction_state
+      @reflects_state ||= []
+      @reflects_state[0] = true
+      super
+    end
+
+    ActiveRecord::Base.__send__(:include, self)
```
Comment (grosser) [inline lib/test_after_commit/with_transaction_state.rb]: same here, use `.include` ?

**Comment by grosser** [comment]
this makes it 2.0.0 only, please add `  s.required_ruby_version = '>= 2.0.0'` to the gemspec

**Comment by alpaca-tc** [comment]
@grosser thx!

File: lib/test_after_commit/with_transaction_state.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# disable parts of the sync code that starts looping
+module TestAfterCommit
+  module WithTransactionState
+    def sync_with_transaction_state
+      @reflects_state ||= []
+      @reflects_state[0] = true
+      super
+    end
+
+    ActiveRecord::Base.prepend(self)
```
Comment (eugeneius) [inline lib/test_after_commit/with_transaction_state.rb]: You're prepending the module after it's required in `test_after_commit.rb`, it doesn't need to be prepended here too.

Comment (grosser) [inline lib/test_after_commit/with_transaction_state.rb]: good catch!

**Comment by grosser** [comment]
manually merged with fix

**Comment by grosser** [comment]
0.5.0

**Comment by grosser** [comment]
hmm this gem should not be used on rails 5 ... was this for silence on 4.2 or are your trying to use it with 5.0 ?

**Comment by eugeneius** [comment]
I found this pull request because I was trying out the Rails 5 beta and saw the deprecation warning. `alias_method_chain` is fine in Rails 4.2, it's only deprecated in Rails 5.

I just noticed that `after_commit` callbacks now run properly during tests on Rails 5 out of the box, so this gem isn't necessary any more - I assume that's what you meant above @grosser.

**Comment by grosser** [comment]
Now blowing up on rails 5 to not confuse people / make them read the readme
:)

**Comment by eugeneius** [comment]
:+1:

**Pull request merged**